### ISSUE TITLE
Constant imported from C header triggers `Integer literal overflows when stored into...`

### ISSUE THREAD

**Issue by jtbandes** [apple/swift#61851]
**Describe the bug**
When a C macro defines an integer constant that overflows its type (intending to produce a negative value from an unsigned literal), importing it to Swift triggers a compiler error when the constant is used.

**Steps To Reproduce**
1. Import CoreFoundation into Swift
2. Attempt to use the `E_NOINTERFACE` value (from CFPlugInCOM.h) in Swift code

**Expected behavior**
`E_NOINTERFACE` can be used.

**Actual behavior**
Compiler error: `Integer literal '2147483652' overflows when stored into signed 'Builtin.Int32'`

Technically, this error is right, but it makes the constant impossible to use in Swift, when it would have worked fine in C/Obj-C.

**Environment**
 - OS: macOS 13.0
 - Xcode Version: 14.0.1 (14A400)

**Additional context**
Note that the C defines are as follows:
```c
typedef SInt32 HRESULT;
#define E_NOINTERFACE ((HRESULT)0x80000004L)
```

**Comment by tbkka** [comment]
Try this:
```
let E_NOINTERFACE = Int32(bitPattern: 0x80000004)
```

**Comment by jtbandes** [comment]
Yes, that would work as a workaround. However, I don't control the header in this case (it comes from a system framework) but I would still like to use the imported constant without defining my own.
